File: 3788-maximum-unique-subarray-sum-after-deletion/3788-maximum-unique-subarray-sum-after-deletion.go
package main

import "sort"

func maxSum(nums []int) int {
	ma := maxElement(nums)
	if ma <= 0 {
		return ma
	}

	us := make(map[int]bool)
	res := 0

	for _, n := range nums {
		if n > 0 && !us[n] {
			res += n
			us[n] = true
		}
	}

	return res
}

func maxElement(arr []int) int {
	if len(arr) == 0 {
		return 0
	}
	sort.Ints(arr)
	return arr[len(arr)-1]
}

File: 3788-maximum-unique-subarray-sum-after-deletion/3788-maximum-unique-subarray-sum-after-deletion.cpp
class Solution {
public:
    int maxSum(vector<int>& nums) {
        int res = *max_element(begin(nums), end(nums)), n = nums.size(), sum = 0;
        unordered_set<int> us;
        for(int i = 0; i < n; i++) {
            if(us.count(nums[i])) continue;
            us.insert(nums[i]);
            sum += max(0,nums[i]);
        }
        return sum > 0 ? sum : res;
    }
};